tap_outreach/sync.py

```diff
@@ -251,7 +251,7 @@ def process_records(stream, mdata, max_modified, records, filter_field, fks):
                                 'null or `id` field expected for `data` relationship')
 
                         # potential fix for the issue - https://github.com/singer-io/tap-outreach/issues/20
-                        if stream.tap_stream_id not in ["prospects", "events"]:
+                        if stream.tap_stream_id not in ["prospects", "events", "mailings"]:
```

Comment:
If a stream encountering this exception means we have to come in and say "and don't throw for this stream" then why are we throwing this exception at all?

Reply:
We should be considering removing the exception block entirely. We should consider overriding the key value from the relationships as per the Outreach support - 

![image](https://github.com/singer-io/tap-outreach/assets/105213416/b5bb4e1c-f083-436c-ae92-4a2240447013)